README.md
# easy_onboarding

A fully customized easy onboarding screen.

This widget can be used as an onboarding screen, a simple walkthrough of how to use you app

<div style="display:flex">
<img width="355" alt="alert2" src="https://user-images.githubusercontent.com/44444254/99188713-de0dc280-2776-11eb-84ca-325a7fe744aa.gif" width="200">
<img width="355" alt="alert2" src="https://user-images.githubusercontent.com/44444254/99188508-d4d02600-2775-11eb-940e-a1f61ebd9bbe.png" width="200">
<img width="344" alt="alert1" src="https://user-images.githubusercontent.com/44444254/99188586-195bc180-2776-11eb-95d6-6ee7c7b81f50.png" width="200">
<img width="344" alt="alert1" src="https://user-images.githubusercontent.com/44444254/99188590-1b258500-2776-11eb-842c-45508c90e812.png" width="200">

<div/>

<b>Usage : </b>

    EasyOnboarding(
      onStart: () {
        print("getting started ");
      },
      skipButtonColor: Colors.transparent,
      backButtonColor: Colors.blue[900],
      nextButtonColor: Colors.blue[900],
      backgroundColor: Colors.white,
      indicatorSelectedColor: Colors.blue[900],
      indicatorUnselectedColor: Colors.blueGrey,
      startButtonColor: Colors.blue[900],
      nextButtonIcon: Icon(
        Icons.arrow_forward,
        color: Colors.white,
      ),
      skipButtonText: Text(
        'SKIP',
        style: TextStyle(fontSize: 15.0, color: Colors.blue[900]),
      ),
      startButtonText: Text(
        'GETTING STARTED',
        style: TextStyle(
          color: Colors.white,
          fontSize: 15.0,
        ),
      ),
      backButtonIcon: Icon(
        Icons.arrow_back,
        color: Colors.white,
      ),
      children: [
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.phone_android,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'A new onboarding experience.',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Easy to use.",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.thumb_up,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Try it.',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Easy to customize.",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.color_lens,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Fully customizable',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Change colors",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.widgets,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Add your own widgets',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "No restriction",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.cloud_download,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Install easy_onboarding',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Created by Dicksen",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
      ],
    );


CHANGELOG.md
## [1.0.1] - TODO: Add release date.

- TODO: Describe initial release.


lib/easy_onboarding.dart
library easy_onboarding;

import 'package:flutter/material.dart';

class EasyOnboarding extends StatefulWidget {
  final Color backgroundColor;
  final Color bottomBackgroundColor;
  final Color indicatorSelectedColor;
  final Color indicatorUnselectedColor;
  final Color startButtonColor;
  final Color backButtonColor;
  final Icon backButtonIcon;
  final Color nextButtonColor;
  final Icon nextButtonIcon;
  final Function onStart;
  final Text startButtonText;
  final List<Widget> children;
  final Color skipButtonColor;
  final Text skipButtonText;

  EasyOnboarding({
    this.backgroundColor = Colors.white,
    this.bottomBackgroundColor = Colors.white,
    this.indicatorSelectedColor = Colors.red,
    this.indicatorUnselectedColor = Colors.grey,
    this.startButtonColor = Colors.red,
    @required this.startButtonText,
    this.backButtonColor = Colors.red,
    @required this.backButtonIcon,
    this.nextButtonColor = Colors.red,
    @required this.nextButtonIcon,
    @required this.onStart,
    @required this.children,
    this.skipButtonColor,
    @required this.skipButtonText,
  });
  @override
  _EasyOnboardingState createState() => _EasyOnboardingState();
}

class _EasyOnboardingState extends State<EasyOnboarding> {
  final PageController _pageController = PageController(initialPage: 0);

  Widget pageIndexIndicator(bool isCurrentPage) {
    return Container(
      margin: EdgeInsets.symmetric(horizontal: 5.0),
      height: isCurrentPage ? 15.0 : 10.0,
      width: isCurrentPage ? 15.0 : 10.0,
      decoration: BoxDecoration(
        color: isCurrentPage
            ? widget.indicatorSelectedColor
            : widget.indicatorUnselectedColor,
        shape: BoxShape.circle,
      ),
    );
  }

  int _currentIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Container(
      height: MediaQuery.of(context).size.height,
      decoration: BoxDecoration(
        color: widget.backgroundColor,
      ),
      child: Scaffold(
        appBar: AppBar(
          elevation: 0.0,
          backgroundColor: widget.backgroundColor,
          actions: _currentIndex == widget.children.length - 1
              ? null
              : [
                  Padding(
                    padding: const EdgeInsets.all(8.0),
                    child: FlatButton(
                      color: widget.skipButtonColor,
                      onPressed: () {
                        _pageController.animateToPage(
                          widget.children.length - 1,
                          duration: Duration(milliseconds: 100),
                          curve: Curves.linear,
                        );
                      },
                      child: widget.skipButtonText,
                    ),
                  ),
                ],
        ),
        backgroundColor: Colors.transparent,
        body: Stack(
          children: [
            PageView(
              controller: _pageController,
              children: widget.children,
              onPageChanged: (val) {
                setState(() {
                  _currentIndex = val;
                });

                print(_currentIndex);
              },
            ),
            Positioned(
              bottom: 100,
              // left: 150,
              child: Container(
                width: MediaQuery.of(context).size.width,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    for (int i = 0; i < widget.children.length; i++)
                      _currentIndex == i
                          ? pageIndexIndicator(true)
                          : pageIndexIndicator(false)
                  ],
                ),
              ),
            )
          ],
        ),
        bottomSheet: _currentIndex == widget.children.length - 1
            ? SafeArea(
                child: Container(
                  width: MediaQuery.of(context).size.width,
                  child: FlatButton(
                    color: widget.startButtonColor,
                    padding: EdgeInsets.symmetric(
                      vertical: 22.0,
                    ),
                    onPressed: widget.onStart,
                    child: widget.startButtonText,
                  ),
                ),
              )
            : SafeArea(
                child: Container(
                  color: widget.bottomBackgroundColor,
                  padding:
                      EdgeInsets.symmetric(horizontal: 20.0, vertical: 20.0),
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      GestureDetector(
                        onTap: () {
                          _pageController.animateToPage(
                            _currentIndex - 1,
                            duration: Duration(milliseconds: 100),
                            curve: Curves.linear,
                          );
                        },
                        child: Container(
                          padding: EdgeInsets.all(10.0),
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: widget.backButtonColor,
                          ),
                          child: widget.backButtonIcon,
                        ),
                      ),
                      GestureDetector(
                        onTap: () {
                          _pageController.animateToPage(
                            _currentIndex + 1,
                            duration: Duration(milliseconds: 100),
                            curve: Curves.linear,
                          );
                        },
                        child: Container(
                          padding: EdgeInsets.all(10.0),
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: widget.nextButtonColor,
                          ),
                          child: widget.nextButtonIcon,
                        ),
                      ),
                    ],
                  ),
                ),
              ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:easy_onboarding/easy_onboarding.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: HomeScreen(),
    );
  }
}

class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return EasyOnboarding(
      onStart: () {
        print("getting started ");
      },
      skipButtonColor: Colors.transparent,
      backButtonColor: Colors.blue[900],
      nextButtonColor: Colors.blue[900],
      backgroundColor: Colors.white,
      indicatorSelectedColor: Colors.blue[900],
      indicatorUnselectedColor: Colors.blueGrey,
      startButtonColor: Colors.blue[900],
      nextButtonIcon: Icon(
        Icons.arrow_forward,
        color: Colors.white,
      ),
      skipButtonText: Text(
        'SKIP',
        style: TextStyle(fontSize: 15.0, color: Colors.blue[900]),
      ),
      startButtonText: Text(
        'GETTING STARTED',
        style: TextStyle(
          color: Colors.white,
          fontSize: 15.0,
        ),
      ),
      backButtonIcon: Icon(
        Icons.arrow_back,
        color: Colors.white,
      ),
      children: [
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.phone_android,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'A new onboarding experience.',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Easy to use.",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.thumb_up,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Try it.',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Easy to customize.",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.color_lens,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Fully customizable',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Change colors",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.widgets,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Add your own widgets',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "No restriction",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
        Container(
          child: Column(
            children: [
              Container(
                child: Icon(
                  Icons.cloud_download,
                  size: 300,
                  color: Color(0xFF679CF8),
                ),
              ),
              Text(
                'Install easy_onboarding',
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.w900,
                ),
              ),
              Text(
                "Created by Dicksen",
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),
      ],
    );
  }
}
